pkg/cache/fair_sharing.go

```diff
@@ -38,19 +38,18 @@ type dominantResourceShareNode interface {
 	hierarchicalResourceNode
 }
 
-// dominantResourceShare returns a value from 0 to 1,000,000 representing the maximum of the ratios
-// of usage above nominal quota to the lendable resources in the cohort, among all the resources
-// provided by the ClusterQueue, and divided by the weight.
-// If zero, it means that the usage of the ClusterQueue is below the nominal quota.
-// The function also returns the resource name that yielded this value.
-// Also for a weight of zero, this will return 9223372036854775807.
+// dominantResourceShare returns a value from 0 to 1,000,000
+// representing the maximum of the ratios of usage above nominal quota
+// to the lendable resources in the cohort, among all the resources
+// provided by the ClusterQueue, and divided by the weight.  If zero,
+// it means that the usage of the ClusterQueue is below the nominal
+// quota.  The function also returns the resource name that yielded
+// this value.  When the FairSharing weight is 0, and the ClusterQueue
+// or Cohort is borrowing, we return math.MaxInt.
 func dominantResourceShare(node dominantResourceShareNode, wlReq resources.FlavorResourceQuantities) (int, corev1.ResourceName) {
 	if !node.HasParent() {
 		return 0, ""
 	}
-	if node.fairWeight().IsZero() {
-		return math.MaxInt, ""
-	}
 
 	borrowing := make(map[corev1.ResourceName]int64, len(node.getResourceNode().SubtreeQuota))
```

Comment:
what if the number of`borrowing` is 0?
Are there still the same problems?

I meant I guess lines 61-62 still have problems.

```
	if len(borrowing) == 0 {
		return 0, ""
	}
```

Reply:
Ah, I missunderstand something. Line 61-62 indicates there is not borrowing. So, we do not need to case the weight value.

/approve

Reply:
/cherry-pick release-0.11